pipeline/dags/parallel_download.py

```diff
@@ -0,0 +1,198 @@
+
+        # This is how it would preferrably be done:
+        # expand() only accepts lists or dicts, so dc_identifiers need to be
+        # converted to a list. With the current implementation, this would
+        # be an issue for large sets, but we have an upcoming ticket to
+        # download huge sets in batches anyway, which should hopefully fix
+        # this issue.
+        # dc_identifiers = list(api.dc_identifiers(set_id))
+
+        @task(
+            task_id=f"download_binding", task_group=download, trigger_rule=trigger_rule
+        )
+        def download_binding(dc_identifier):
+            binding_id = utils.binding_id_from_dc(dc_identifier)
+            ssh_hook = SSHHook(ssh_conn_id=ssh_conn_id)
+            with ssh_hook.get_conn() as ssh_client:
+                sftp_client = ssh_client.open_sftp()
+
+                utils.make_intermediate_dirs(
+                    sftp_client=sftp_client,
+                    remote_directory=f"{base_path}/{set_id.replace(':', '_')}/{binding_id}/mets",
+                )
+
+                SaveMetsSFTPOperator(
+                    task_id=f"save_mets_{binding_id}",
+                    api=api,
+                    sftp_client=sftp_client,
+                    dc_identifier=dc_identifier,
+                    base_path=base_path,
+                    file_dir=f"{set_id.replace(':', '_')}/{binding_id}/mets",
+                ).execute(context={})
+
+                SaveAltosForMetsSFTPOperator(
+                    task_id=f"save_altos_{binding_id}",
+                    sftp_client=sftp_client,
+                    base_path=base_path,
+                    file_dir=f"{set_id.replace(':', '_')}/{binding_id}/alto",
+                    mets_path=f"{base_path}/{set_id.replace(':', '_')}/{binding_id}/mets",
+                    dc_identifier=dc_identifier,
+                ).execute(context={})
+
+        download_binding.expand(dc_identifier=dc_identifiers)
+
+    return download
+
+
+with DAG(
+    dag_id="parallel_download",
+    schedule="@once",
+    catchup=False,
+    default_args=default_args,
+    doc_md=__doc__,
+) as dag:
+
+    create_nlf_connection = CreateConnectionOperator(
+        task_id="create_nlf_connection",
+        conn_id="nlf_http_conn",
+        conn_type="HTTP",
+        host="https://digi.kansalliskirjasto.fi/interfaces/OAI-PMH",
+        schema="HTTPS",
+    )
+
+    begin_download = EmptyOperator(task_id="begin_download")
+
+    cancel_pipeline = EmptyOperator(task_id="cancel_pipeline")
+
+    @task.branch(task_id="check_api_availability")
+    def check_api_availability():
+        """
+        Check if API is responding and download can begin. If not, cancel pipeline.
+        """
+        api_ok = HttpSensor(
+            task_id="http_sensor", http_conn_id="nlf_http_conn", endpoint="/"
+        ).poke(context={})
+
+        if api_ok == True:
+            return "begin_download"
+        elif api_ok == False:
+            return "cancel_pipeline"
```

Comment:
Could we just do `if api_ok:` accompanied by `else`? Or if it's important that False is different from None, then `if api_ok is True` + `elif api_ok is False` is more pythonic.

Reply:
I was first weary of cases where the poking function may decide to not return anything, but now that I think about it we probably want to cancel the pipeline in those cases as well (and it shouldn't ever happen anyway), so `if api_ok:` and `else` should work just fine.